### PR TITLE
ENT-5039 Improved help text for commands

### PR DESCRIPTION
**Pull request by jakubbielawa**
Improved help text for most of the commands.

### CODE REVIEW

**Review by nickdunstone13 — DISMISSED** [review]

File: tools/shell/src/main/java/net/corda/tools/shell/StartShellCommand.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
     @Command
     @Man("An alias for 'flow start'. Example: \"start Yo target: Some other company\"")
+    @Usage("An alias for 'flow start'. Example: \"start Yo target: Some other company\"")
```
Comment (blsemo) [inline tools/shell/src/main/java/net/corda/tools/shell/StartShellCommand.java]: Not sure this does actually get shown:
```
Fri Feb 28 09:47:35 GMT 2020>>> start
You must pass a name for the flow. Example: "start Yo target: Some other company"
```

File: tools/shell/src/main/java/net/corda/tools/shell/HashLookupShellCommand.java [hunk 2]
```diff
@@ -24,11 +24,20 @@
             "state is claimed to be already consumed by another transaction.\n\n" +
             "Example usage: hashLookup E470FD8A6350A74217B0A99EA5FB71F091C84C64AD0DE0E72ECC10421D03AAC9"
     )
+    @Usage("Checks if a transaction matching a specified Id hash value is recorded on this node.\n\n" +
+            "This is mainly intended to be used for troubleshooting notarisation issues when a\n" +
+            "state is claimed to be already consumed by another transaction.\n\n" +
+            "Example usage: hashLookup E470FD8A6350A74217B0A99EA5FB71F091C84C64AD0DE0E72ECC10421D03AAC9"
+    )
```
Comment (blsemo) [inline tools/shell/src/main/java/net/corda/tools/shell/HashLookupShellCommand.java]: I am not quite sure how one would go about to see this help text?

File: tools/shell/src/main/java/net/corda/tools/shell/FlowShellCommand.java [hunk 3]
```diff
@@ -31,7 +31,11 @@
     private static final Logger logger = LoggerFactory.getLogger(FlowShellCommand.class);
 
     @Command
-    @Usage("Start a (work)flow on the node. This is how you can change the ledger.")
+    @Usage("Start a (work)flow on the node. This is how you can change the ledger.\n\n" +
+            "\t\t    Starting flow is the primary way in which you command the node to change the ledger.\n" +
+            "\t\t    This command is generic, so the right way to use it depends on the flow you wish to start. You can use the 'flow start'\n" +
+            "\t\t    command with either a full class name, or a substring of the class name that's unambiguous. The parameters to the\n" +
+            "\t\t    flow constructors (the right one is picked automatically) are then specified using the same syntax as for the run command.\n")
```
Comment (blsemo) [inline tools/shell/src/main/java/net/corda/tools/shell/FlowShellCommand.java:38]: The lines are a bit long for displaying in a shell.

**Review by blsemo — COMMENTED** [review]
I have run the node from this branch and had a go at looking what the messages look like in the shell - this is definitely an improvement. See a few comments inline.

File: tools/shell/src/main/java/net/corda/tools/shell/StartShellCommand.java (hunk 1 above)
Comment (jakubbielawa) [reply tools/shell/src/main/java/net/corda/tools/shell/StartShellCommand.java]: You're right, I removed it. It's not necessary.

File: tools/shell/src/main/java/net/corda/tools/shell/HashLookupShellCommand.java (hunk 2 above)
Comment (jakubbielawa) [reply tools/shell/src/main/java/net/corda/tools/shell/HashLookupShellCommand.java]: Removed this as well as it turned out to be unnecessary.

File: tools/shell/src/main/java/net/corda/tools/shell/FlowShellCommand.java (hunk 3 above)
Comment (jakubbielawa) [reply tools/shell/src/main/java/net/corda/tools/shell/FlowShellCommand.java:38]: We'd like to leave it like this as this is what's shown by the 'man' command in the unsafe mode and it explains how to use 'flow' very well.

**Comment by blsemo** [comment]
I don't know how this usage thing works in detail in the shell, but it would be good if `start` would show a bit more help what you need to provide to make it work. I remember being thouroughly confused by this when I had just joined.

**Comment by jakubbielawa** [comment]
> I don't know how this usage thing works in detail in the shell, but it would be good if `start` would show a bit more help what you need to provide to make it work. I remember being thouroughly confused by this when I had just joined.

@blsemo For some reason I'm not able to respond directly to your comment. I agree, it would be great if `start` showed a bit more about how to use it but I think that should be a separate change since my changes only make sure the text that has already existed for the `man` command gets displayed for `help`.

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**